src/script/interpreter.cpp

```diff
@@ -1106,8 +1110,64 @@ class CTransactionSignatureSerializer {
 
 } // anon namespace
 
-uint256 SignatureHash(const CScript& scriptCode, const CTransaction& txTo, unsigned int nIn, int nHashType)
+uint256 SignatureHash(const CScript& scriptCode, const CTransaction& txTo, unsigned int nIn, int nHashType, const CAmount& amount, int sigversion)
 {
+    if (sigversion == 1) {
+        uint256 hashPrevouts;
+        uint256 hashSequence;
+        uint256 hashOutputs;
+
+        if (!(nHashType & SIGHASH_ANYONECANPAY)) {
+            CHashWriter ss(SER_GETHASH, 0);
+            for (unsigned int n = 0; n < txTo.vin.size(); n++) {
+                ss << txTo.vin[n].prevout;
+            }
+            hashPrevouts = ss.GetHash(); // TODO: cache this value for all signatures in a transaction
+        }
+
+        if (!(nHashType & SIGHASH_ANYONECANPAY) && (nHashType & 0x1f) != SIGHASH_SINGLE && (nHashType & 0x1f) != SIGHASH_NONE) {
```

Comment:
Right, I had written that off as vestigial since I don't see standard mempool policy being changed to accept non-final transactions.
As a signer, why would switching from caring about all outputs to caring only about a subset of outputs make me not care about other input sequence numbers?
When is that useful?
The reason for keeping this would have to be for some offline protocol. But, I can't think of one where this actually works out being necessary.
Also, this coupled with BIP68 allows a counterparty to change the time at which their signature becomes usable (i.e.- when the transaction becomes BIP68 final and relayable).
This seems benign, but if I care about other inputs, why would I not care about this effect?